### PR TITLE
Query: Avoid pushdown when joining with a subquery which can be lifte…

### PR DESCRIPTION
**Pull request by smitpatel**
…d out

Part of #17622

Avoids pushdown for scenario when we cause pushdown due to order by but then we erase order by without skip take giving us simple and lift-able query again

### CODE REVIEW

**Comment by smitpatel** [comment]
@dotnet/efteam - ping

**Review by maumar — APPROVED** [review]

File: test/EFCore.SqlServer.FunctionalTests/Query/ComplexNavigationsCollectionsQuerySqlServerTest.cs [hunk 1]
```diff
@@ -1096,37 +1096,31 @@ public override async Task Select_subquery_single_nested_subquery(bool async)
     ) AS [t]
     WHERE [t].[row] <= 1
 ) AS [t0] ON [l].[Id] = [t0].[OneToMany_Optional_Inverse2Id]
-LEFT JOIN (
-    SELECT [l1].[Id], [l1].[OneToMany_Optional_Inverse3Id]
-    FROM [LevelThree] AS [l1]
-) AS [t1] ON [t0].[Id] = [t1].[OneToMany_Optional_Inverse3Id]
-ORDER BY [l].[Id], [t0].[Id], [t1].[Id]");
+LEFT JOIN [LevelThree] AS [l1] ON [t0].[Id] = [l1].[OneToMany_Optional_Inverse3Id]
```
Comment (roji) [inline test/EFCore.SqlServer.FunctionalTests/Query/ComplexNavigationsCollectionsQuerySqlServerTest.cs:1099]: :heart: very cool, thanks for doing this @smitpatel!

Comment (NinoFloris) [reply test/EFCore.SqlServer.FunctionalTests/Query/ComplexNavigationsCollectionsQuerySqlServerTest.cs:1099]: I love how implementing such a change has this ripple effect on queries, opening up other simplifications in turn.

Comment (roji) [reply test/EFCore.SqlServer.FunctionalTests/Query/ComplexNavigationsCollectionsQuerySqlServerTest.cs:1099]: Yeah, it's kind of like how inlining in real compilation opens the way for other optimization types... Not sure we have good examples of that in EF Core, but :rofl:

**Pull request merged**